shadowsocks/udp.go

```diff
@@ -34,12 +34,19 @@ import (
 
 const udpBufSize = 64 * 1024
 
-// Wrapper for logger.Debugf during UDP access key searches.
-func debugUDP(cipherID, template string, val interface{}) {
+// Wrapper for logger.Debugf during UDP proxying.
+func debugUDP(tag string, template string, val interface{}) {
 	// This is an optimization to reduce unnecessary allocations due to an interaction
 	// between Go's inlining/escape analysis and varargs functions like logger.Debugf.
 	if logger.IsEnabledFor(logging.DEBUG) {
-		logger.Debugf("UDP(%s): "+template, cipherID, val)
+		logger.Debugf("UDP(%s): "+template, tag, val)
+	}
+}
+
+func debugUDPAddr(addr net.Addr, template string, val interface{}) {
```

Comment:
I don't think this method is needed. According to https://golang.org/pkg/fmt/#pkg-overview, the String() method will be called automatically. We just need to stop calling it, as far as I understand.

Reply:
I'm not sure what you're suggesting, but note that the first argument to `debugUDP` is a `string`.  Changing that argument to `interface{}`, as would be required to pass a `net.Addr`, causes an allocation for every `id` in `unpack`, e.g. 100 allocations in BenchmarkUDPUnpackFail.